### ISSUE TITLE
Upload app to AppStore

### ISSUE THREAD

**Issue by Yosifz8** [youtube/youtube-ios-player-helper#399]
I'm trying to upload a YouTube app using this framework, But the Apple review team keep reject the app stating that it violates their rules:

`5.2.3 Audio/Video Downloading: Apps should not facilitate illegal file sharing or include the ability to save, convert, or download media from third-party sources (e.g. Apple Music, YouTube, SoundCloud, Vimeo, etc.) without explicit authorization from those sources. Streaming of audio/video content may also violate Terms of Use, so be sure to check before your app accesses those services. Documentation must be provided upon request.`

Did you have this problem? Know where I can get the Documentation from YouTube?

Thanks!

**Comment by acosmicflamingo** [comment]
What is the nature of your app? Does your app give absolutely any impression that users can access the YouTube videos offline? Do you explicitly mention that this framework is the one that YouTube officially endorses: https://developers.google.com/youtube/v3/guides/ios_youtube_helper?

**Comment by Frankenxtein** [comment]
Any update on this? I am receiving the same rejection.

**Comment by Yosifz8** [comment]
> What is the nature of your app? Does your app give absolutely any impression that users can access the YouTube videos offline? Do you explicitly mention that this framework is the one that YouTube officially endorses: https://developers.google.com/youtube/v3/guides/ios_youtube_helper?

Just playing a video from Youtube, online streaming.



> Any update on this? I am receiving the same rejection.

No update, still in the same issue.

**Comment by vaibhiarora03** [comment]
Any update on this? Did anyone got any solution to this? I am also served with the same rejection.

**Comment by andrew-skrypnyk** [comment]
@vaibhiarora03 @Yosifz8 @Frankenxtein guys, did you find the solution?

**Comment by vaibhiarora03** [comment]
@andrew-skrypnyk After around 8 rejections, I finally removed the YouTube player view and streamed the videos using a url with videos hosted with AWS.

**Comment by EdJones** [comment]
@Yosifz8, @Frankenxtein, @vaibhiarora03, @andrew-skrypnyk did you resubmit and ask the human reviewers?

**Comment by vaibhiarora03** [comment]
Yes, I resubmitted the app and mentioned the details in notes for the review team.

**Comment by denkeni** [comment]
Here's the good news: App Store Review now approves 3rd party apps with YouTube streaming. Just include **YouTube API Services Terms of Service <https://developers.google.com/youtube/terms/api-services-terms-of-service>** in the review notes.

I brought this issue to App Review Lab in WWDC21, and the reviewer confirmed that current internal policy has been updated to accept terms of service, and suggested me to resubmit it again. I did resubmit, and my app [Playlist Note](https://apps.apple.com/app/id1332640775) has been approved with YouTube content again, through another 2 weeks in review including 1 rejection, though.

Other side notes:
- Allowing content access without any limitation (i.e. without filtering objectionable keywords), Age Rating must be raised to 17+.
- Other streaming services might still be rejected ;)

One silly thing is that even MusicKit from Apple Music was rejected. One app reviewer said “the documentation for Apple Music isn't sufficient for streaming full content of the songs”, so I needed to point out that Apple Developer Program License Agreement 3.3.50 (MusicKit section) stated: 
> - You may play MusicKit Content only as rendered by the MusicKit APIs or MusicKit JS and only as permitted in the Documentation (e.g., album art and music-related text from the MusicKit API may not be used separately from music playback or managing playlists)  

Clarifying that our app implemented MusicKit APIs from Apple and did exactly what were permitted.

**Comment by gregorybesson** [comment]
Thank you so much @denkeni for your insights. I've been refused until I see your post and follow your advices.

My app, Punchline : sounds fun is now on the app store.

I use the official youtube-player-helper with this awesome project for swiftui : https://github.com/SvenTiigi/YouTubePlayerKit

**Comment by tronglv94** [comment]
I'm facing with rejection from Apple Review, even I had included the YouTube API Services Terms of Service https://developers.google.com/youtube/terms/api-services-terms-of-service in the review notes. Can anyone here help me? plz.

Here is their response:
"The issues we previously identified still need your attention before we can approve your submission.
The documentation you have provided does not sufficiently demonstrate permission to represent the products or services included in your app, or the parties identified in the documentation are not correct. We are unable to continue with the review until we have verified that you are currently authorized by the rights holder to represent the products or services available in your app."

**Comment by andrew-skrypnyk** [comment]
In addition to YouTube agreement, we shared our agreements with authors, who we work with and Apple approved our app

**Comment by devsigofficial** [comment]
**App Store Review now approves 3rd party apps with YouTube streaming. Just include YouTube API Services Terms of Service https://developers.google.com/youtube/terms/api-services-terms-of-service in the review notes.**
